payme: init at 1.2.0

cimm opened the pull request:
## Description of changes

[payme](https://github.com/jovandeginste/payme), a command line QR code generator for SEPA payments, written in golang.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

cimm committed:
payme: init at 1.2.0

AndersonTorres commented on pkgs/by-name/pa/payme/package.nix:
```diff
@@ -0,0 +1,23 @@
+
+buildGoModule rec {
+  pname = "payme";
+  version = "v1.2.0";
```
```suggestion
  version = "1.2.0";
```

`version` should start with a number.

AndersonTorres commented on pkgs/by-name/pa/payme/package.nix:
```diff
@@ -0,0 +1,23 @@
+
+  src = fetchFromGitHub {
+    owner = "jovandeginste";
+    repo = pname;
```
```suggestion
    repo = "payme";
```

Do not parameterize `pname`.

AndersonTorres commented on pkgs/by-name/pa/payme/package.nix:
```diff
@@ -0,0 +1,23 @@
+
+  vendorHash = null;
+
+  meta = with lib; {
```
```suggestion
  meta = {
```

Nested with is bad.

AndersonTorres commented on pkgs/by-name/pa/payme/package.nix:
```diff
@@ -0,0 +1,23 @@
+  meta = with lib; {
+    description = "QR code generator (ASCII & PNG) for SEPA payments";
+    homepage = "https://github.com/jovandeginste/payme";
+    license = licenses.mit;
+    maintainers = with maintainers; [ cimm ];
+    platforms = with lib.platforms; unix;
```
```suggestion
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ cimm ];
```

`platforms` is unneeded here - since it use nothing besides Go

AndersonTorres commented:
Now squash.

cimm commented:
Thank you for the useful feedback @AndersonTorres! I have added the requested changes and squashed the commits.

I am not sure yet why it is bad to nest `with lib`, I have done it in other packages as well so this is something I will look into and learn. Thanks for pointing that out!

AndersonTorres commented:
Here are some pointers about the issues with `with`:

https://github.com/NixOS/nixpkgs/issues/208242

AndersonTorres reviewed (approved):
Result of `nixpkgs-review pr 292826` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>payme</li>
  </ul>
</details>
